### PR TITLE
Implement package watcher

### PR DESCRIPTION
**Pull request by mtojek**
This PR introduces a package watcher that can be used in dev mode.

Thanks to this update, packages are served from memory cache if there are no important changes in the filesystem.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -100,6 +102,8 @@ func main() {
 	if err := server.Shutdown(ctx); err != nil {
 		log.Fatal(err)
 	}
+
+	util.ClosePackageWatcher()
```
Comment (ruflin) [inline main.go:106]: What about using a `defer util.ClosePackageWatcher()` on line 77 instead? Like this we can keep the logic together and the watcher is not closed when it is not started.

File: util/package_watcher.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	"github.com/pkg/errors"
+	"github.com/radovskyb/watcher"
+)
+
+const watcherPollingPeriod = time.Second
+
+var (
+	w *watcher.Watcher
+
+	watchedPackages []Package
```
Comment (ruflin) [inline util/package_watcher.go:21]: Why not use the same package list cache from packages.go? It seems the other one is not used if this is enabled.

File: util/package_watcher.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+	for _, p := range packagePaths {
+		err = w.AddRecursive(p)
+		if err != nil && !os.IsNotExist(err) {
+			log.Fatal(errors.Wrapf(err, "watching directory failed (path: %s)", p))
+		}
+	}
+
+	go func() {
+		go w.Start(watcherPollingPeriod)
```
Comment (ruflin) [inline util/package_watcher.go:44]: Why is this started inside the go routine? Could this go on line 42?

Comment (mtojek) [reply util/package_watcher.go:44]: Actually, it blocks. I can move the whole line up: `go w.Start(...)`

Comment (ruflin) [reply util/package_watcher.go:44]: jep, that is what I mean. I just saw routine in routine which set off alarm bells :-)

File: util/package_watcher.go (hunk 2 above)
Comment (mtojek) [reply util/package_watcher.go]: Fixed.

File: main.go (hunk 1 above)
Comment (mtojek) [reply main.go]: Fixed.

File: util/package_watcher.go (hunk 3 above)
Comment (mtojek) [reply util/package_watcher.go]: Fixed :)

**Pull request merged**